exporter/azuredataexplorerexporter/factory.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuredataexplorerexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuredataexplorerexporter"
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr            = "azuredataexplorer"
+	managedingesttype  = "managed"
+	queuedingesttest   = "queued"
+	unknown            = "unknown"
+	defaultmetrictable = "OTELMetrics"
+	defaultlogtable    = "OTELLogs"
+	defaulttracetable  = "OTELTraces"
+	metricstype        = 1
+	logstype           = 2
+	tracestype         = 3
+)
+
+// Creates a factory for the ADX Exporter
+func NewFactory() component.ExporterFactory {
+	return component.NewExporterFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithTracesExporter(createTracesExporter),
+		component.WithMetricsExporter(createMetricsExporter),
+		component.WithLogsExporter(createLogsExporter),
+	)
+}
+
+/*Create default configurations*/
+func createDefaultConfig() config.Exporter {
+	return &Config{
+		ClusterUri:      "https://CLUSTER.kusto.windows.net",
+		ApplicationId:   unknown,
```

Comment:
Question: wouldn't having the string "unknown" pass config.Validate? how do we keep this from happening? Or will this never reach this flow?

Reply:
I guess we need to drop it from the defaultConfig creation, Else it will pass the validation and will get error once the credential validation happens.